src/Composer/DependencyResolver/DefaultPolicy.php

```diff
@@ -57,7 +57,7 @@ public function versionCompare(PackageInterface $a, PackageInterface $b, string
         }
 
         // dev versions need to be compared as branches via matchSpecific's special treatment, the rest can be optimized with compiling matcher
-        if (strpos($a->getVersion(), 'dev-') === 0 || strpos($b->getVersion(), 'dev-') === 0) {
+        if ($a->isDev() || $b->isDev()) {
```

Comment:
isDev uses VersionParser::parseStability which also matches -dev suffix. So I don't think this is strictly equivalent, and may cause issues when comparing 1.999.999-dev to 2.999.999.x-dev for example. It probably won't match incorrectly, but it removes the CompilingMatcher optimization, so you could argue it's making it a bit slower too 🤷🏻 

Can you benchmark this?

```suggestion
        if (($a->isDev() && strpos($a->getVersion(), 'dev-') === 0) || ($b->isDev() && strpos($b->getVersion(), 'dev-') === 0)) {
```